yt_dlp/extractor/franceculture.py

```diff
@@ -61,65 +31,68 @@ def _real_extract(self, url):
-            webpage, 'playlist data', fatal=False, default=None)
-
-        if playlist_data:
-            entries = []
-            for item, item_description in re.findall(
-                    r'(?s)(<button[^<]*class="[^"]*replay-button[^>]*>).*?<p[^>]*class="[^"]*teaser-text-chapo[^>]*>(.*?)</p>',
-                    playlist_data):
-
-                item_attributes = extract_attributes(item)
-                entries.append({
-                    'id': item_attributes.get('data-emission-uuid'),
-                    'url': item_attributes.get('data-url'),
-                    'title': item_attributes.get('data-diffusion-title'),
-                    'duration': int_or_none(traverse_obj(item_attributes, 'data-duration-seconds', 'data-duration-seconds')),
-                    'description': item_description,
-                    'timestamp': int_or_none(item_attributes.get('data-start-time')),
-                    'thumbnail': info['thumbnail'],
-                    'uploader': info['uploader'],
-                })
-
-            return {
-                '_type': 'playlist',
-                'entries': entries,
-                **info
-            }
+        video_data = json.loads(
+            self._search_regex(  # type: ignore
+                r'({"@type":"AudioObject","contentUrl":"[^"]+","duration":"[^"]+","encodingFormat":"mp3","potentialAction":{"@type":"Action","name":"ListenAction"}})',
+                webpage,
+                'video data',
+            )
+        )
+        video_url = video_data['contentUrl']
+        ext = video_data['encodingFormat']
 
-        video_data = extract_attributes(self._search_regex(
-            r'''(?sx)
-                (?:
-                    </h1>|
-                    <div[^>]+class="[^"]*?(?:title-zone-diffusion|heading-zone-(?:wrapper|player-button))[^"]*?"[^>]*>
-                ).*?
-                (<button[^>]+data-(?:url|asset-source)="[^"]+"[^>]+>)
-            ''',
-            webpage, 'video data'))
-        video_url = traverse_obj(video_data, 'data-url', 'data-asset-source')
-        ext = determine_ext(video_url.lower())
+        # 'P0Y0M0DT0H38M28S'
+        duration = sum(
+            [
+                durations[0] * durations[1]
+                for durations in zip(
+                    [
+                        int(duration)
+                        for duration in re.split(r'[PYMDTOHMS]', video_data['duration'])
+                        if duration
+                    ],
+                    [
+                        365 * 60 * 60 * 24,  # how many seconds in a year
+                        30 * 60 * 60 * 24,  # how many seconds in a month
+                        60 * 60 * 24,  # how many seconds in a day
+                        60 * 60,  # how many seconds in an hour
+                        60,  # how many second an in minute
+                        1,  # how many seconds in a second
+                    ],
+                )
+            ]
+        )
```

Comment:
(actually this should be implemented in `parse_duration`)

Reply:
I'm having a hard time trying to figure out which helper functions can be used, and what they are doing. It seems my (non-existing) knowledge of the codebase prevents me to contribute in a way that's satisfying :confused: Can someone help?

Reply:
Implemented ✅